paparazzi/src/main/java/app/cash/paparazzi/internal/PixelPerfect.kt

```diff
@@ -78,7 +78,13 @@ internal object PixelPerfect : Differ {
 
     // 3 different colors, 256 color levels
     val total = actualHeight.toLong() * actualWidth.toLong() * 3L * 256L
-    val percentDifference = (delta * 100 / total.toDouble()).toFloat()
+    var percentDifference = (delta * 100 / total.toDouble()).toFloat()
+
+    // If the delta diff is all black pixels, the computed difference is 0, but there are still
+    // different pixels. We can fallback to the amount of different pixels to less precise difference to ensure difference is reported.
+    if (differentPixels > 0 && percentDifference == 0f) {
+      percentDifference = differentPixels * 100 / (actualWidth * actualHeight.toDouble()).toFloat()
+    }
```

Comment:
nit: it looks like we can avoid the `var` here and above, if the comparison uses `delta` (the numerator in percentDifference) instead.

```suggestion
    // If the delta diff is all black pixels, the computed difference is 0, but there are still
    // different pixels. We can fallback to the amount of different pixels to less precise difference to ensure difference is reported.
    val percentDifference = if (delta == 0 && differentPixels > 0) {
differentPixels * 100 / (actualWidth * actualHeight.toDouble()).toFloat()
    } else {
      (delta * 100 / total.toDouble()).toFloat()
    }
```